subsys/dfu/dfu_target/src/dfu_target_smp.c

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <pm_config.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/dfu/mcuboot.h>
+#include <zephyr/mgmt/mcumgr/mgmt/mgmt.h>
+#include <zephyr/mgmt/mcumgr/smp/smp_client.h>
+#include <zephyr/mgmt/mcumgr/grp/img_mgmt/img_mgmt_client.h>
+#include <zephyr/mgmt/mcumgr/grp/os_mgmt/os_mgmt.h>
+#include <zephyr/mgmt/mcumgr/grp/os_mgmt/os_mgmt_client.h>
+#include <dfu/dfu_target.h>
+#include <dfu/dfu_target_smp.h>
+#include <zephyr/devicetree.h>
+
+LOG_MODULE_REGISTER(dfu_target_smp, CONFIG_DFU_TARGET_LOG_LEVEL);
+
+#define MCUBOOT_HEADER_MAGIC 0x96f3b83d
```

Comment:
Include `bootutil/image.h` and use `IMAGE_MAGIC`

Reply:
Build system not find that header